utils/rag/rag_components.py

```diff
@@ -332,27 +611,53 @@ def generate_subquestions(self, state):
 
         return {"subquestions": subquestions}
     
-    def detect_entities(self, state):
+    def detect_entities(self, state: Dict) -> Dict:
+        """
+        Detects entities in a given question or subquestion.
 
-        print("---DETERMINING ENTITIES---")
-        question = state["question"]
-        subquestions = state["subquestions"]
+        Args:
+            state (Dict): The input state dict.
 
-        if len(subquestions) == 0:
-            response = self.entity_chain.invoke({"question": question})
+        Returns:
+            state (Dict): The updated state dict containing the detected entities.
+        """
 
-        else:
-            response = self.entity_chain.invoke({"question": subquestions[0]})
+        print("---DETERMINING ENTITIES---")
+        question: str = state["question"]
+        # print("question: ", question)
+        subquestions: List[str] = state["subquestions"]
 
-        entities = response["entity_name"].split()
-        entities = [e.lower() for e in entities]
-        entities = [e.replace(",", "") for e in entities]
-        entities = [e + ".pdf" for e in entities] # Dumb hack to avoid dealing with the vectorstore logic, for now.
-        print(entities)
+        try:
+            if len(subquestions) == 0:
+                response: Dict = self.entity_chain.invoke({"question": question})
+
+            else:
+                response: Dict = self.entity_chain.invoke({"question": subquestions[0]})
```

Comment:
Are we sure that this is always given by the first element?

Reply:
I always pop(0) for the first subquestion when answering, so yep - entity should always be for the question that's going to get popped off later.  Nice catch, though.